pkg/chain/evmclient.go

```diff
@@ -69,6 +61,27 @@ func (ec *evmClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) er
 	return errors.Wrap(multierr.Combine(errs...), "batch call error")
 }
 
+func (ec *evmClient) OnNewHead(ctx context.Context, cb func(blockKey types.BlockKey)) error {
+	ch := make(chan *ethtypes.Header, 1)
+	sub, err := ec.SubscribeNewHead(ctx, ch)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-sub.Err():
+			if sub, err = ec.SubscribeNewHead(ctx, ch); err != nil {
+				return err
+			}
+		case head := <-ch:
+			cb(types.BlockKey(head.Number.String()))
+		}
+	}
+}
```

Comment:
q for my understanding: in prod this code is not actually used, we use the evm client from main CL repo right?

Reply:
This is the wrapper that allows subscribing on new heads in more chain-agnostic way

Reply:
What i wanted to confirm was that `NewClient` does not seem to be called anywhere in this repo other than mocks / tests. Is that correct?

Reply:
NewEVMClient could be used by plugin's users, i.e. chainlink node.

Reply:
@infiloop2 some of these constructors are used in the `chainlink` repo to build and pass dependencies to the plugin.

Reply:
It helps separate chain specific logic from the plugin.